### DISCUSSION TITLE
strongswan-5.9.3dr2 IKE1 IPsec SA: only UDP encapsulation is supported failed to create SAD entry

### DISCUSSION THREAD

**Discussion by whbwyj** [strongswan/strongswan#1838] [General]
Server ipsec.conf configuration
# ipsec.conf - strongSwan IPsec configuration file
# basic configuration
conn %default
	ikelifetime=1440m
	keylife=60m
	rekeymargin=3m
	keyingtries=0
	keyexchange=ikev1  #ike版本
	authby=secret
conn n2n
	left=192.168.245.128
	leftid=192.168.245.128  #本端公网IP
	leftsubnet=192.168.142.0/24  #本端私有网络地址 客户端配置虚拟IP，服务器用0.0.0.0/0
	right=192.168.245.139  #对端公网IP  %any 客户端ID，什么都行
	rightsubnet=192.168.72.0/24  #对端私有网络地址
	auto=start  #进程主动时立即建立IPsec安全连接
	type=tunnel
	ike=3des-md5-modp1024
	esp=3des-md5
config setup
	# strictcrlpolicy=yes
	# uniqueids = no

client ipsec.conf configuration
# ipsec.conf - strongSwan IPsec configuration file
# basic configuration
conn %default
        ikelifetime=1440m
        keylife=60m
        rekeymargin=3m
        keyingtries=0
        keyexchange=ikev1  #ike版本
        authby=secret
conn n2n
        left=192.168.245.139
        leftid=192.168.245.139  #本端公网IP
        leftsubnet=192.168.72.0/24  #本端私有网络地址
        right=192.168.245.128  #对端公网IP
        rightsubnet=192.168.142.0/24  #对端私有网络地址
        auto=start  #进程主动时立即建立IPsec安全连接
        type=tunnel
        ike=3des-md5-modp1024
        esp=3des-md5

Enable IPSec separately, and the client reports to the server:
/ipsec/sbin/ipsec up n2n
generating QUICK_MODE request 2610924643 [ HASH SA No ID ID ]
sending packet: from 192.168.245.139[500] to 192.168.245.128[500] (220 bytes)
received packet: from 192.168.245.128[500] to 192.168.245.139[500] (172 bytes)
parsed QUICK_MODE response 2610924643 [ HASH SA No ID ID ]
selected proposal: ESP:3DES_CBC/HMAC_MD5_96/NO_EXT_SEQ
  IPsec SA: only UDP encapsulation is supported
failed to create SAD entry
  IPsec SA: only UDP encapsulation is supported
failed to create SAD entry
unable to install inbound and outbound IPsec SA (SAD) in kernel
generating INFORMATIONAL_V1 request 2848941558 [ HASH N(NO_PROP) ]
sending packet: from 192.168.245.139[500] to 192.168.245.128[500] (68 bytes)
establishing connection 'n2n' failed

**Comment by strongX509** [comment]
You seem to use the **`kernel-libipsec`** plugin which with strongSwan versions < 5.9.11 supports ESP-in-UDP encapsulation only. You have to force UDP encapsulation by adding the **`forceencaps=yes`** option in **`ipsec.conf`**.

**Reply by tobiasbrunner** [reply]
Note that this is enforced automatically by the plugin. The problem is, however, that forcing UDP encapsulation only works if the peer supports NAT-T, which might not be the case with some IKEv1 peers.
